proxy/logging/LogConfig.cc

```diff
@@ -837,19 +893,21 @@ LogConfig::update_space_used()
               "The rolled logfile, %s, was auto-deleted; "
               "%" PRId64 " bytes were reclaimed.",
               candidates[victim].name, candidates[victim].size);
+
+        // update space after successful unlink
         m_space_used -= candidates[victim].size;
         m_partition_space_left += candidates[victim].size;
+        target->total_size -= candidates[victim].size;
       }
+      // Update total candidates and victim
+      total_candidate_count--;
+      victim++;
     }
   }
-  //
-  // Clean up the candidate array
-  //
-  for (i = 0; i < candidate_count; i++) {
-    ats_free(candidates[i].name);
-  }
 
-  //
+  for (auto iter = deleting_info.begin(); iter != deleting_info.end(); iter++) {
```

Comment:
If you're using `IntrusiveHashMap`, `IntrusiveHashMap::apply` is a better choice. Also, should you be deleting the entries in the table?